cli/scorecard/score.go

```diff
@@ -233,7 +233,8 @@ func (inventory *InventoryConfig) Score(config *ScoringConfig, outputPath string
 			for _, category := range config.categories {
 				for _, cv := range category.constraints {
 					for _, v := range cv.Violations {
-						record := []string{category.Name, getConstraintShortName(v.Constraint), v.Resource, v.Message}
+						parent := v.Metadata.GetStructValue().Fields["details"].GetStructValue().Fields["asset"].GetStructValue().Fields["ancestors"].GetListValue().Values[0].GetStringValue()
```

Comment:
I built from branch but getting a nil pointer dereference with the[ bad bucket example](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/feat/scorecard-project/cli/docs/scorecard.md#scorecard-user-guide). 

Not super familiar with the codebase but `v.Metadata.GetStructValue().Fields["details"]` only seems to have `resource` field. 
 
```
Generating CFT scorecard
WARNING: Unable to read inventory file : org_policy_inventory.json storage: object doesn't exist
WARNING: Unable to read inventory file : access_policy_inventory.json storage: object doesn't exist
Category,Constraint,Resource,Message,Parent
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x215b5ac]

goroutine 1 [running]:
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/scorecard.(*InventoryConfig).Score(0xc0005f0420, 0xc0021e2048, 0x0, 0x0, 0x7ffeefbffa54, 0x3, 0x30d10f0, 0x0, 0x0, 0x0, ...)
        /Users/PATH/cloud-foundation-toolkit/cli/scorecard/score.go:236 +0x12ec
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/scorecard.glob..func2(0x307b8a0, 0xc0005be300, 0x0, 0x4, 0x0, 0x0)
        /Users/PATH/cloud-foundation-toolkit/cli/scorecard/cmd.go:108 +0x398
github.com/spf13/cobra.(*Command).execute(0x307b8a0, 0xc0005be2c0, 0x4, 0x4, 0x307b8a0, 0xc0005be2c0)
        /Users/bbaiju/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x307aea0, 0x1001570, 0x0, 0x0)
        /Users/bbaiju/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/bbaiju/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/GoogleCloudPlatform/cloud-foundation-toolkit/cli/cmd.Execute()
        /Users/PATH/cloud-foundation-toolkit/cli/cmd/root.go:75 +0x2d
main.main()
        /Users/PATH/cloud-foundation-toolkit/cli/main.go:7 +0x25
 ```

Reply:
I added a check and also some tests to cover this.

Reply:
@morgante tested with the same example. I expected the bad bucket to have a parent. IIUC that was the original ask?
```
Category,Constraint,Resource,Message,Parent
Security,denylist_public_users,//storage.googleapis.com/MY_BAD_BUCKET,//storage.googleapis.com/MY_BAD_BUCKET is publicly accessable,
```

Reply:
@bharathkkb It turns out I was looking in the wrong place (because *some* constraints included the asset in metadata). PTAL at the updated code, with new test cases.